src/codegen/llvm/codegen_llvm_visitor.hpp

```diff
@@ -45,8 +49,18 @@ class CodegenLLVMVisitor: public visitor::ConstAstVisitor {
     // Output directory for code generation
     std::string output_dir;
 
-    // result string for demo
-    std::string result_code;
+  private:
+    std::unique_ptr<llvm::LLVMContext> context = std::make_unique<llvm::LLVMContext>();
+
+    std::unique_ptr<llvm::Module> module = std::make_unique<llvm::Module>(mod_filename, *context);
```

Comment:
@georgemitenkov : I am re-reading this again and I am wondering if this shouldn't be in the constructor? because, mod_filename here is not initialised yet here? unless I am missing something obvious.

Reply:
When we call the constructor, we initialise `mod_file`, and therefore implicitly initialise `module` with default call `std::make_unique<llvm::Module>(mod_filename, *context)` - based on C++11 this is ok.

Putting this into constructor may make more sense (all initialisation in one place), but that's just refactoring/preference. I think that this file will be updated as more codegen will be supported - like custom module dump routines, documentation, etc.

Reply:
Reference: Usage section in https://en.cppreference.com/w/cpp/language/data_members

Reply:
Ah...! thanks for reminding that! all good!